cocotb/handle.py

```diff
@@ -692,20 +692,28 @@ def _set_value(self, value, call_sim):
         because assigning integers less than 32 bits is faster.
 
         Args:
-            value (ctypes.Structure, cocotb.binary.BinaryValue, int, double):
+            value (cocotb.binary.BinaryValue, int):
                 The value to drive onto the simulator object.
 
         Raises:
             TypeError: If target is not wide enough or has an unsupported type
                  for value assignment.
+
+        .. deprecated:: 1.5
+            :class:`ctypes.Structure` objects are no longer accepted as values for assignment.
+            Use ``int.from_bytes(bytes(struct_obj), byteorder='big')`` instead; see :meth:`int.from_bytes`.
         """
         value, set_action = self._check_for_set_action(value)
 
         if isinstance(value, int) and value < 0x7fffffff and len(self) <= 32:
             call_sim(self, self._handle.set_signal_val_long, set_action, value)
             return
         if isinstance(value, ctypes.Structure):
-            value = BinaryValue(value=cocotb.utils.pack(value), n_bits=len(self))
+            warnings.warn(
+                "`ctypes.Structure` values are no longer accepted for value assignment. "
+                "Use `int.from_bytes(bytes(struct_obj), byteorder='big')` instead",
```

Comment:
I assume you've checked that `big` is equivalent to the old behavior here?

Reply:
I might be tempted to just recommend `BinaryValue(value=bytes(struct_obj), n_bits=len(signal))`, since I have no idea if what you suggest behaves the same for structs of the wrong size.

Reply:
`bigEndian=True` is default for `BinaryValue`, but `BinaryValue` probably does impart some funkiness on the whole thing.